llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -16433,7 +16433,9 @@ bool AArch64TargetLowering::lowerDeinterleaveIntrinsicToLoad(
   if (UseScalable && !VTy->isScalableTy())
     return false;
 
-  unsigned NumLoads = getNumInterleavedAccesses(VTy, DL, UseScalable);
+  // Get around for the missing of split large size scalarize vectors.
+  // TODO: Support the custom split of large size scalarize vectors
+  unsigned NumLoads = getNumInterleavedAccesses(VTy, DL, false);
```

Comment:
This looks like it would break the SVE fixed length support.  Looking at the code I think changing:
```if (UseScalable)```
to
```if (UseScalable && isa<FixedVectorType>(VecTy)``` 
should do the trick given `getMinSVEVectorSizeInBits()` is not relevant for scalable vector types and the default `128` value should be maintained.

Reply:
Apply your comment, thanks @paulwalker-arm

Reply:
Sorry, I should have been clearer.  I meant the code within `getNumInterleavedAccesses()` needs to be changed because that is where the bug exists.